### PR TITLE
refactor to dockerize app

### PR DESCRIPTION
**Pull request by krisdo**
-added Dockerfile
-added docker-compose.yml
-added wait-for-it.sh
-added a script to combine seeding and starting the server in package json

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - .:/home/node/app
+      - node_modules:/home/node/app/node_modules
+    # networks:
+    #   - app-network
+    ports:
+      - "2001:2001"
+    depends_on:
+      - "mongo"
+    command: bash -c "/usr/wait-for-it.sh --timeout=0 mongo:27017 && npm run build"
+volumes:
```
Comment (samasastudio) [inline docker-compose.yml:41]: I believe if you're calling your wait-for-it script here then it's probably redundant to run it in your Dockerfile as well.  See what overlap you might have and if your compose still works after taking some of those overlapping parts away.

File: database-mongodb/index.js [hunk 2]
```diff
@@ -4,10 +4,31 @@ module.exports.mongoose = mongoose;
...
+};
+
+//for docker
+mongoose.connect(url, options)
+.then( function() {
+  console.log('MongoDB is connected');
+})
+.catch( function(err) {
+  console.log(err);
 });
```
Comment (samasastudio) [inline database-mongodb/index.js:33]: I'm going to try this myself, but for running the app in development without containers, I think you can set up fallback hosts so you don't have to comment in/out connects for your database.

File: Dockerfile [hunk 3]
```diff
@@ -3,11 +3,11 @@ FROM node:10
...
 # why the *
-COPY package*.json /usr/src/
+COPY package.json .
 RUN npm install
 
 COPY . .
 
 COPY wait-for-it.sh /usr/wait-for-it.sh
-RUN chmod +x /usr/wait-for-it.sh
+# RUN chmod +x /usr/wait-for-it.sh
```
Comment (cmonez) [inline Dockerfile:12]: Hmmm. I should try commenting this out. I just read that this was necessary in order to run the bash command.

Comment (krisdo) [reply Dockerfile:12]: Same, it's to make it executable. However, by addressing bash -c that could be diff?

**Pull request merged**